### PR TITLE
6540 Adds mixed content block back end

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

:warning:  based on #11575 

This PR adds a new 'mixed content block' consisting of a listing and a video block. 

image

Link to sample test page: after running `inv new-env` the block will be available on mozfest.localhost:8000/en/
Related PRs/issues: https://torchbox.monday.com/boards/1334760528/pulses/1337896540

# Checklist

**Tests**
- [ ] Is the code I'm adding covered by tests?

**Changes in Models:**
- [x] Did I update or add new fake data?
- [ ] Did I squash my migration?
- [ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?

**Documentation:**
- [ ] Is my code documented?
- [ ] Did I update the READMEs or wagtail documentation?

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

### CODE REVIEW

File: network-api/networkapi/mozfest/blocks/mixed_content_block.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        help_text="Text content to display withe video title.",
+    )
+    thumbnail = image_blocks.ImageChooserBlock(help_text="The image to show before the video is played.")
+
+    class Meta:
+        icon = "media"
+
+
+class MixedContentBlock(blocks.StructBlock):
+    video = blocks.ListBlock(VideoBlock(), max_num=1, required=False)
```
Comment (kevinhowbrook) [inline network-api/networkapi/mozfest/blocks/mixed_content_block.py:39]: needs unwrapping from list block

File: network-api/networkapi/mozfest/blocks/mixed_content_block.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
+from django.forms.utils import ErrorList
+from wagtail import blocks
+from wagtail.blocks import struct_block
+from wagtail.images import blocks as image_blocks
+
+from networkapi.wagtailpages.pagemodels.customblocks import listing as listing_blocks
+
+
+class VideoBlock(blocks.StructBlock):
+    url = blocks.CharBlock(
```
Comment (tbrlpld) [inline network-api/networkapi/mozfest/blocks/mixed_content_block.py:10]: Is it on purpose that this is not using a `URLBlock`?

Comment (tbrlpld) [reply network-api/networkapi/mozfest/blocks/mixed_content_block.py:10]: Or an `EmbedBlock`?

Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/blocks/mixed_content_block.py]: I went off the other video_block I found, but this should be a URL block at least yea, I've changed it to that. Chris is doing the FE at the moment so I'm unsure as to whether we want this to be and embed of just a URL value tbh

File: network-api/networkapi/mozfest/blocks/mixed_content_block.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        help_text="Text content to display withe video title.",
+    )
+    thumbnail = image_blocks.ImageChooserBlock(help_text="The image to show before the video is played.")
+
+    class Meta:
+        icon = "media"
+
+
+class MixedContentBlock(blocks.StructBlock):
+    video = VideoBlock(max_num=1)
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/blocks/mixed_content_block.py]: nah, thanks, removed

File: network-api/networkapi/mozfest/blocks/mixed_content_block.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
+from django.forms.utils import ErrorList
+from wagtail import blocks
+from wagtail.blocks import struct_block
+from wagtail.images import blocks as image_blocks
+
+from networkapi.wagtailpages.pagemodels.customblocks import listing as listing_blocks
+
+
+class VideoBlock(blocks.StructBlock):
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/blocks/mixed_content_block.py:9]: TBC, I'll wait for chris to finish his FE template work. Fwiw though it doesn't for now as it's only used as part of this mixed content block and isn't configured to be added anywhere else individually.

File: network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                            {% if item.link_page %}
+                                <a href="{% pageurl item.link_page %}"
+                                   class="tw-group/listing hover:tw-no-underline focus:tw-no-underline hover:tw-text-black focus:tw-text-black"
+                                   aria-label="{{ item.title }}">
+                            {% elif item.link_url %}
+                                <a href="{{ item.link_url }}" target="_blank" rel="noopener"
+                                   class="tw-group/listing hover:tw-no-underline focus:tw-no-underline hover:tw-text-black focus:tw-text-black"
+                                   aria-label="{{ item.title }}">
+                            {% endif %}
+                            <img width="auto" height="auto" src="{{ img.url }}" alt="{{ item.alt_text }}"/>
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html:25]: Yes it works, the image tag is used above this:

```
                        {% image item.image fill-350x197 as img %}
```

**Comment by kevinhowbrook** [comment]
Having cherry-picked chris's FE work into this locally, it all looks fine so I'll wait for that review

File: network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html [hunk 6]
```diff
@@ -0,0 +1,44 @@
+{% load wagtailcore_tags wagtailimages_tags %}
+
+{% block block_content %}
+    <div class="tw-py-16">
+        <div class="tw-container">
+            <div class="tw-row tw-justify-center">
+                {{ self.video }}
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html]: yea it's required 
image

File: network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                <h2 class="tw-h3-heading">{{ self.video.title }}</h2>
+                <div class="[&_p,li,ul]:last:tw-mb-0">
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html]: Looks like chris removed this before merging in

**Comment by kevinhowbrook** [comment]
RE the EmdedBlock question, looks like the existing videos all use URL values, and the only EmbedBlock I see is DatawrapperBlock so I don't want to break the already adopted convention there

File: network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html (hunk 6 above)
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/mozfest/templates/fragments/blocks/mixed_content_block.html]: it does if you are depending on the individual block template, but we just depend on the one mixed content template for this as we aren't supporting the individual video block

**Pull request merged**